OTVision/transform/transform.py

```diff
@@ -21,51 +21,66 @@
 import geopandas as gpd
 import numpy as np
+import pandas as pd
 
 from OTVision.config import CONFIG
-from OTVision.helpers.files import get_files, read_json, write_json
+from OTVision.helpers.files import (
+    _check_and_update_metadata_inplace,
+    get_files,
+    read_json,
+    replace_filetype,
+    write_json,
+)
 from OTVision.helpers.formats import (
     _get_datetime_from_filename,
     _get_epsg_from_utm_zone,
-    _get_fps_from_filename,
     _get_time_from_frame_number,
     _ottrk_dict_to_df,
 )
-from OTVision.helpers.log import log
+from OTVision.helpers.log import log, reset_debug, set_debug
 
 from .get_homography import get_homography
 
 
 def main(
-    paths,
-    refpts_file=None,
-    overwrite=CONFIG["TRANSFORM"]["OVERWRITE"],
+    paths: list[Path],
+    refpts_file: Union[Path, None] = None,
+    overwrite: bool = CONFIG["TRANSFORM"]["OVERWRITE"],
     debug: bool = CONFIG["TRANSFORM"]["DEBUG"],
-):
-    """Transforms .ottrk file containing trajectories in pixel coordinates
-    to .gpkg file with trajectories in utm coordinates
-    using either one single .otrfpts file for all .ottrk files
-    containing corresponding reference points in both pixel and utm coordinates
-    or using specific .otrfpts files for each .ottrk file
+) -> None:
+    """Transform tracks files containing trajectories in pixel coordinates to .gpkg
+    files with trajectories in utm coordinates using either one single refpts file for
+    all tracks files containing corresponding reference points in both pixel and utm
+    coordinates or using specific refpts files for each tracks file
     (path must be the same except for the extension).
-
-    Info: .otrfpts file can be created using OTVision.transform.reference_points_picker
+    Info: refpts file can be created using OTVision.transform.reference_points_picker
 
     Args:
-        tracks_files (str or Path or list): (List of) Path(s) to .ottrk file
-        single_refpts_file (str or Path, optional): Path to .otrfpts file. Default: None
+        paths (list[Path]): List of paths to tracks files.
+        refpts_file (Path, optional): Path to reference points file.
+            If given, this file will be used for transformation of all tracks files.
+            If not given, for every tracks file a refpts file with same stem is needed.
+            Defaults to None.
+        overwrite (bool, optional): Whether or not to overwrite existing tracks files in
+            world coordinates.
+            Defaults to CONFIG["TRANSFORM"]["OVERWRITE"].
+        debug (bool, optional): Whether or not to run in debug mode.
+            Defaults to CONFIG["TRANSFORM"]["DEBUG"].
     """
 
     log.info("Start transformation from pixel to world coordinates")
     if debug:
-        log.setLevel("DEBUG")
-        log.debug("Debug mode on")
+        set_debug()
 
     if refpts_file:
+        refpts_file = get_files(
+            paths=[refpts_file], filetypes=CONFIG["FILETYPES"]["REFPTS"]
```

Comment:
What happens, if more than one refpts files can be found?
The assumption is that there is only one, right? But how do we determine the correct one when several could be found?

This is not that urgent for this PR and can be addressed later. Best to document it.

Reply:
I think if we pass only one file path to get files with only one possible filetype, only one refpts_file can be found. But yes, it is kind of ugly right now. We have to improve this and also the filetypes in the config

Reply:
I noted this for later